contracts/FeralfileArtworkV4_1.sol

```diff
@@ -0,0 +1,147 @@
+
+import "./FeralfileArtworkV4.sol";
+
+contract FeralfileExhibitionV4_1 is FeralfileExhibitionV4 {
+    struct AdvancedData {
+        address artist;
+        uint256 amount;
+    }
+
+    mapping(address => uint256) public artistAdvancedAmounts;
+
+    constructor(
+        string memory name_,
+        string memory symbol_,
+        bool burnable_,
+        bool bridgeable_,
+        address signer_,
+        address vault_,
+        address costReceiver_,
+        string memory contractURI_,
+        uint256[] memory seriesIds_,
+        uint256[] memory seriesMaxSupplies_,
+        AdvancedData[] memory data_
+    )
+        FeralfileExhibitionV4(
+            name_,
+            symbol_,
+            burnable_,
+            bridgeable_,
+            signer_,
+            vault_,
+            costReceiver_,
+            contractURI_,
+            seriesIds_,
+            seriesMaxSupplies_
+        )
+    {
+        for (uint256 i = 0; i < data_.length; i++) {
+            artistAdvancedAmounts[data_[i].artist] = data_[i].amount;
+        }
+    }
+
+    /// @notice pay to get artworks to a destination address. The pricing, costs and other details is included in the saleData
+    /// @param r_ - part of signature for validating parameters integrity
+    /// @param s_ - part of signature for validating parameters integrity
+    /// @param v_ - part of signature for validating parameters integrity
+    /// @param saleData_ - the sale data
+    function buyArtworks(
+        bytes32 r_,
+        bytes32 s_,
+        uint8 v_,
+        SaleData calldata saleData_
+    ) external payable override {
+        require(_selling, "FeralfileExhibitionV4: sale is not started");
+        super._checkContractOwnedToken();
+        validateSaleData(saleData_);
+
+        saleData_.payByVaultContract
+            ? vault.payForSale(r_, s_, v_, saleData_)
+            : require(
+                saleData_.price == msg.value,
+                "FeralfileExhibitionV4: invalid payment amount"
+            );
+
+        bytes32 message = keccak256(
+            abi.encode(block.chainid, address(this), saleData_)
+        );
+
+        require(
+            isValidSignature(message, r_, s_, v_),
+            "FeralfileExhibitionV4: invalid signature"
+        );
+
+        uint256 itemRevenue;
+        if (saleData_.price > saleData_.cost) {
+            itemRevenue =
+                (saleData_.price - saleData_.cost) /
+                saleData_.tokenIds.length;
+        }
+
+        uint256 distributedRevenue;
+        uint256 platformRevenue;
+        for (uint256 i = 0; i < saleData_.tokenIds.length; i++) {
+            // send NFT
+            _safeTransfer(
+                address(this),
+                saleData_.destination,
+                saleData_.tokenIds[i],
+                ""
+            );
+            // distribute royalty
+            RevenueShare[] memory revenueShares = saleData_.revenueShares[i];
+            bool isArtist = false;
```

Comment:
Those variable `isArtist` and the `artistAddr` could be unify as one, it looks like redundant.